src/app/autocomplete/autocomplete.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, Output, EventEmitter, OnDestroy } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { FormControl } from '@angular/forms';
+import { HttpClient } from '@angular/common/http';
+import {GeoJSON} from 'geojson';
+import { MatOptionSelectionChange } from '@angular/material/core';
+
+@Component({
+  selector: 'app-autocomplete',
+  templateUrl: './autocomplete.component.html',
+  styleUrls: ['./autocomplete.component.css']
+})
+export class AutocompleteComponent implements OnDestroy {
+  YOUR_API_KEY = "";
+  @Output()
+  locationChange: EventEmitter<PlaceSuggestion> = new EventEmitter<PlaceSuggestion>();
+  searchOptions: Subject<PlaceSuggestion[]> = new Subject<PlaceSuggestion[]>();
+  inputFieldFormControl: FormControl = new FormControl();
```

Comment:
Just a note - it's better to give your outputs an alias in case refactoring occurs - approving for now.